rackcity/views/it_model_views.py

```diff
@@ -21,6 +21,21 @@ def model_list(request):
     #         return JsonResponse(serializer.data, status=201)
     #     return JsonResponse(serializer.errors, status=400)
 
+def model_add(request): # TODO: make it only accept POSTs
+    """
+    Add a new model
+    """
+    if request.method == 'POST':
+        data = JSONParser().parse(request)
+        serializer = ITModelSerializer(data=data)
+        if serializer.is_valid(): # TODO: figure out what .valid() does.  Different fields are required for add, update, etc
```

Comment:
There's more logic that should be done here.  For example, checking that the vendor+model_num is unique.  It should also allow you to leave model_id blank and fill it in itself (maybe that should even be required?)

Reply:
Maybe each endpoint should have a function that encapsulates its validation and returns usable errors when validation fails?

Reply:
Good idea. Also we can enforce the `model_id` field to be unique on the Django model (as an argument on the field), but I don't think the front end should have to pass that in with this POST, since it's always going to be concatenated, so I think we would want to do that in this method?

Reply:
On second thought...I don't think we need `model_id` at all since it's sort of redundant. We can use this on `vendor` and `model_number`

https://docs.djangoproject.com/en/dev/ref/models/options/#unique-together